README.md

```diff
@@ -118,6 +118,66 @@ following cmake options:
     -DTEST_COMPILER_IN_DOCKER_DEBIAN_TAG="bookworm"
     -DTEST_COMPILER_IN_DOCKER_GCC_VERSION="12"
 
+Developing in QEMU
+---------------
+
+Sometimes it may be useful to run sio2jail in a virtual machine, e.g. to test running
+sio2jail under a different kernel version. In QEMU:
+
+    # get debian 11 QEMU image
+    wget https://cdimage.debian.org/images/cloud/bullseye/latest/debian-11-nocloud-amd64.qcow2
+
+    # reserve more space for QEMU image
+    qemu-img resize debian-11-nocloud-amd64.qcow2 +10G
+
+    # start QEMU with KVM and sio2jail sources mounted with virtfs
+    qemu-system-x86_64 \
+        -enable-kvm \
+        -m 4G \
+        -nographic \
+        -hda debian-11-nocloud-amd64.qcow2 \
+        -virtfs local,path=$SIO2JAIL_SOURCES,security_model=mapped-xattr,mount_tag=sio2jail \
+        -cpu max
+
+    # login as root (no password)
+
+    # resize root partition
+    apt update
+    apt install -y cloud-utils fdisk
+    growpart /dev/sda 1
+    resize2fs /dev/sda1
+
+    # install dependencies and utilities
+    apt install -y \
+        build-essential \
+        cmake \
+        gcc-multilib \
+        g++-multilib \
+        gperf \
+        wget
+
+    # mount sio2jail sources
+    mkdir /sio2jail
+    mount -t 9p -o trans=virtio sio2jail /sio2jail -oversion=9p2000.L,posixacl,msize=104857600
+
+    # enable required sysctls
+    sysctl -w kernel.perf_event_paranoid=-1
+    sysctl -w kernel.unprivileged_userns_clone=1
+
+    # create a separate user with same uid and gid as your local user (check id -u)
+    groupadd -g $GID dev
+    useradd -u $UID -g dev -m dev -s /bin/bash
```

Comment:
can we make this part (except for the mount) a script?
So that this part can be simplified to sth like:
```sh
mkdir /sio2jail
mount -t 9p -o trans=virtio sio2jail /sio2jail -oversion=9p2000.L,posixacl,msize=104857600
bash /sio2jail/qemu/guest-setup.sh
```

?

Or maybe there are tools that would let us script the whole process of interacting with the VM, so that we can later automatically run a test matrix on multiple different kernels?

Reply:
Probably we can... I'll try to script this in the future